### ISSUE TITLE
Accessibility guidelines and legal requirements

### ISSUE THREAD

**Issue by jaguarfi** [apluslms/a-plus#501]
> **Migrated from Trello**

New legislation requires that learning systems at the university are accessible (public governmental service).

Some links (Web accessibility guideline 2.1, ETSI standard EN 301 549):
https://www.w3.org/TR/WCAG21/
https://www.w3.org/WAI/WCAG21/quickref/
http://mandate376.standards.eu/standard

WAVE web accessibility evaluation tool (browser extension)
https://wave.webaim.org/extension/

Examples:

* alt attributes in `<image>`
* aria attributes
* explanations visible only to screen readers (Bootstrap CSS class `sr-only`) when images or colours are used to convey important information (such as warnings)
* using the web pages should not require a mouse, but work only with the keyboard
* All web pages should have clear landmarks in the DOM. HTML5 elements header, main, section, aside, and footer should be used properly. The main element is mandatory and wraps the main content on the page. Navigation must be inside nav elements.

iPhone has a built-in screen reader that is easy to use for developers. It shows visually the part of the page that is being read aloud, and thus one can see if some parts are not working with the screen reader.

**Comment by Mankro** [comment]
Note that we are also required to publish a report about how we implement accessibility requirements (**accessibility statement**, in Finnish *saavutettavuusseloste*). There are templates and tools for writing the statement.

Accessibility statement in English:
https://www.w3.org/WAI/planning/statements/

More informational links from a webinar (in Finnish):

*    https://www.saavutettavuusvaatimukset.fi/saavutettavuusseloste/
*    https://saavutettavuusvaatimukset.fi
*    https://info.digicampus.fi
*    https://info.digicampus.fi/osahankkeet/esteettomyys-ja-saavutettavuus/
*    WebCAG https://www.saavutettavuusvaatimukset.fi/lait-ja-standardit/wcag-2-1/
*    https://www.celia.fi/Blogi/viisi-vinkkia-saavutettavampaan-verkkosisaltoon/
*    http://papunet.net/saavutettavuus
* example report from a government site in English https://finlandabroad.fi/web/finlandabroad/accessibility-statement and in Finnish: https://finlandabroad.fi/saavutettavuusseloste

Tools

* https://wave.webaim.org/ web accessibility evaluation tool
* siteimprove tool https://siteimprove.com/en/accessibility/
* axe automated accessibility testing https://www.deque.com/axe/axe-for-web/

**Comment by Mankro** [comment]
This issue is a meta issue about all accessibility requirements. More detailed, individual accessibility requirements are added as new issues with the label "area: accessibility".

For example, #546 and #559.

**Comment by raphendyr** [comment]
> More detailed, individual accessibility requirements are added as new issues with the label "area: accessibility".

Those issues should reference this issue in their description, so they leave a mark in the timeline here. The status of those issues is then clearly visible here.

**Comment by Mankro** [comment]
Accessibility statement / saavutettavuusseloste was added #652
